pkgs/development/python-modules/pyqtgraph/default.nix

```diff
@@ -1,26 +1,58 @@
 { lib
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
+, scipy
 , numpy
-, pyopengl
 , pyqt5
-, scipy
+, pyopengl
+, qt5
+, python
+, pytestCheckHook
+, freefont_ttf
+, makeFontsConf
+, fetchpatch
 }:
 
+let
+  fontsConf = makeFontsConf {
+    fontDirectories = [ freefont_ttf ];
+  };
+in
 buildPythonPackage rec {
   pname = "pyqtgraph";
-  version = "0.12.1";
+  version = "0.12.2";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "d2ef3b5289184fb48dfe5f44ccb58d9d64ffb5452fc524a2bd7a640a36b3874d";
+  src = fetchFromGitHub {
+    owner = "pyqtgraph";
+    repo = "pyqtgraph";
+    rev = "pyqtgraph-${version}";
+    sha256 = "093kkxwj75nb508vz7px4x7lxrwpaff10pl15m4h74hjwyvbsg3d";
   };
 
-  propagatedBuildInputs = [ numpy pyopengl pyqt5 scipy ];
+  # TODO: remove when updating to 0.12.3
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/pyqtgraph/pyqtgraph/commit/2de5cd78da92b48e48255be2f41ae332cf8bb675.patch";
+      sha256 = "1hy86psqyl6ipvbg23zvackkd6f7ajs6qll0mbs0x2zmrj92hk00";
+    })
+  ];
+
+  propagatedBuildInputs = [ numpy pyqt5 scipy pyopengl ];
+
+  checkInputs = [ pytestCheckHook ];
 
-  doCheck = false;  # tries to create windows (QApps) on collection, which fails (probably due to no display)
+  checkPhase = ''
+    export QT_PLUGIN_PATH="${qt5.qtbase.bin}/${qt5.qtbase.qtPluginPrefix}"
+    export QT_QPA_PLATFORM=offscreen
+    export DYLD_FRAMEWORK_PATH=/System/Library/Frameworks
+    export FONTCONFIG_FILE=${fontsConf}
 
-  pythonImportsCheck = [ "pyqtgraph" ];
+    # disable 6 tests which try to clone https://github.com/pyqtgraph/test-data
+    ${python.interpreter} test.py --pyqt5 \
+      -k "not test_ImageItem and not test_ImageItem_axisorder \
+      and not test_PlotCurveItem and not test_getArrayRegion \
+      and not test_getArrayRegion_axisorder and not test_PolyLineROI"
```

Comment:
Does pytestCheckHook not pick up the test entrypoint correctly, or why do you call it here manually?

If you want to disable tests, add them to `disabledTests`. If you need arbitrary arguments added to the pytest call, use `pytestFlagsArray`.

Reply:
Thanks for the review

i checked and they dont need to be disabled anymore since the data is included in the repo